config/filter.d/apache-modsecurity.conf

```diff
@@ -10,7 +10,7 @@ before = apache-common.conf
 [Definition]
 
 
-failregex = ^%(_apache_error_client)s ModSecurity:\s+(?:\[(?:\w+ \"[^\"]*\"|[^\]]*)\]\s*)*Access denied with code [45]\d\d
+failregex = ^%(_apache_error_client)s *ModSecurity:\s+(?:\[(?:\w+ \"[^\"]*\"|[^\]]*)\]\s*)*Access denied with code [45]\d\d
```

Comment:
Hmm...
Note that quantifier `*` applied to the token after it is placed, so in this case for space ` ` only.

May be you want something like this...
```diff
- failregex = ^%(_apache_error_client)s ModSecurity:\s+(?:\[(?:\w+ \"[^\"]*\"|[^\]]*)\]\s*)*Access denied with code [45]\d\d
+ failregex = ^%(_apache_error_client)s(?: \[client [^\]]+\])? ModSecurity:\s+(?:\[(?:\w+ \"[^\"]*\"|[^\]]*)\]\s*)*Access denied with code [45]\d\d
```

Reply:
There are two client instances next to each other in some versions of mod_security (see my log entry).  The * would catch both of them

Reply:
Nope, it would not - the `*` is after space, so it matches 0..N spaces only.
I try to explain this - you see, the second case does not match:
```bash
$ python -c 'import re; m = re.match(r"^C * mod", "C mod"); print(m.group(0) if m else m)'
C mod
$ python -c 'import re; m = re.match(r"^C * mod", "C C mod"); print(m.group(0) if m else m)'
None
$ python -c 'import re; m = re.match(r"^C(?: C)? mod", "C C mod"); print(m.group(0) if m else m)'
C C mod
```
Possibly you've tried with `*` directly after `%(_apache_error_client)s` (so again, not after space), then it can work, but it is too aggressive...

Reply:
Yes you are right, thank you